rdmo/core/xml.py

```diff
@@ -257,6 +257,21 @@ def validate_pre_conversion_for_missing_key_in_legacy_elements(elements, version
             raise ValueError(f"Missing legacy elements, elements containing 'key' were expected for this XML with version {version} and elements {models_in_elements}.")   # noqa: E501
 
 
+def update_related_elements(elements: Dict, related_uri: str, related_model: str, related_key: str, field_name: str):
```

Comment:
What ist this method used for?

Reply:
I think there was something broken with the current legacy xml parsing because the related elements are not written back into the ascending elements (following the current element models).
For example, for the sections I added:
```py
del element['catalog']  # sections do not have catalog anymore
```
This is correct right? They don't have `catalog` anymore..
But then the corresponding catalog needs to know about this section so I've added this function to "update" the catalog in that case.
The same then holds for `pages`, `questionsets`, `questions` etc.. right?

Still it's for the legacy xml's.. I could also revert if needed...